indigo_app/middlware.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.messages import get_messages
+import json
+from datetime import datetime, timedelta
+from django.template.loader import render_to_string
+class HtmxMessagesMiddleware:
+    """ Adds a message to the Django messages framework if the request is an htmx request.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        # The HX-Request header indicates that the request was made with HTMX
+        if "HX-Request" not in request.headers:
+            return response
+
+        # Ignore redirections because HTMX cannot read the headers
+        if 300 <= response.status_code < 400:
+            return response
+
+        # check for messages
+        messages = get_messages(request)
+        if not messages:
+            return response
+
+        response.write(
```

Comment:
what's this doing? appending to the end of the response? what if it's not an HTML response?

Reply:
hmm, hadn't thought of that, let me change it up. There is another approach that uses HX-Trigger header that may be more suitable.

Reply:
another option is, for now, just to add this logic into the one template that needs it -- the work update form. then we make it generic when we need to do so. that will unblock here and let you move forward.

Reply:
I've updated this to use the HX-Trigger header, I had tested this before changing it out for the previous approach. I thought injecting the toast into the response was simpler because you don't have any javascript.  I suppose I hadn't thought it through completely. This approach is more suitable for our situation I think.

https://www.loom.com/share/185f00a5635945b788d5a99ba7296c4b?sid=a43373f2-a01f-4de6-be16-17df30409dbc